ophyd/utils/paths.py

```diff
@@ -0,0 +1,71 @@
+import os
+import calendar
+
+
+def makedirs(path, *, mode=0o777, mode_base=None):
+    '''Recursively make directories and set permissions
+
+    Parameters
+    ----------
+    path : str
+        Full path to create, including all parent directories
+    mode : int, optional
+        Mode to set
+    mode_base : str, optional
+        If specified, only try to set mode after this directory
+        Full base path
+    '''
+    # Permissions not working with os.makedirs -
+    # See: http://stackoverflow.com/questions/5231901
+    if not path or os.path.exists(path):
+        return []
+
+    head, tail = os.path.split(path)
+    ret = makedirs(head, mode=mode, mode_base=mode_base)
+    try:
+        os.mkdir(path)
```

Comment:
that is what the `os.chmod` a few lines down fixes.  I think you can let python do the recursion checking for you and then juts fix up the permissions?